.github/workflows/ci.yml

```diff
@@ -154,41 +248,25 @@ jobs:
       pull-requests: write
     steps:
       - uses: actions/checkout@v4
-      - name: Download PR performance test results
+      - name: Download base ref performance test results
         uses: actions/download-artifact@v4
         with:
-          pattern: '*-${{ github.sha }}-iperf3results'
+          pattern: '*-${{ github.event.pull_request.base.sha }}-iperf3results'
           merge-multiple: true
-          path: ./
-      - name: Get last Continous Delivery workflow run Id
-        uses: actions/github-script@v7
-        id: get_last_cd_run
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          script: |
-            const { data } = await github.rest.actions.listWorkflowRuns({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              workflow_id: "cd.yml",
-              status: 'success',
-              per_page: 1
-            });
-            return data.workflow_runs[0].id;
-      - name: Download main branch performance test results
+          path: ./${{ github.event.pull_request.base.sha }}
+      - name: Download head ref performance test results
```

Comment:
So if this workflow is never run on `main`, how do we get results to compare?

Reply:
We run performance tests on the `base_ref` which is more correct for cases where we've stacked the PR.

Reply:
Same trick I use to test DB migrations in CI:

- Checkout base_ref
- Run steps
- Check head_ref
- Re-run steps

It's a little bit of duplication but worth it IMO to avoid linking workflow runs together across branches which makes CI brittle.

Reply:
So we always run them twice per PR? Once for the base ref and once for the current one? Would it be worth to extract that into a workflow then that we call with two different refs? I think that would make things easier to understand.

Feel free to do in a follow-up if you want to get this merged.

Reply:
I think those two could also run in parallel which would be nice.

Reply:
Ah yes, good call they should just be a matrix job

Reply:
Updated, this will merge when it passes.